staging/src/k8s.io/apiserver/pkg/audit/policy/dynamic.go

```diff
@@ -37,3 +38,17 @@ func ConvertDynamicPolicyToInternal(p *v1alpha1.Policy) *audit.Policy {
 		OmitStages: InvertStages(stages),
 	}
 }
+
+// NewDynamicChecker returns a new dynamic policy checker
+func NewDynamicChecker() Checker {
+	return &dynamicPolicyChecker{}
+}
+
+type dynamicPolicyChecker struct{}
```

Comment:
I'd prefer to keep this simple for the alpha. As in, just plop a featuregate check around this line: https://github.com/kubernetes/kubernetes/blob/94c5953904f4f365776eb2c61b6b28d95f005d4c/staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go#L123, with the "audit everything" approach inlined.

Reply:
This made it somewhat simpler to build the policy checker and keep the behavior consistent in the handler. Today passing a nil checker bypasses the handler. I also plan on expanding on this to help with the full event serialization issue. That being said not entirely opposed, just want to be clear about the reasoning